src/Features/CSharp/Portable/ConvertAnonymousTypeToTuple/CSharpConvertAnonymousTypeToTupleDiagnosticAnalyzer.cs

```diff
@@ -14,8 +17,10 @@ internal class CSharpConvertAnonymousTypeToTupleDiagnosticAnalyzer
             SyntaxKind,
             AnonymousObjectCreationExpressionSyntax>
     {
-        protected override SyntaxKind GetAnonymousObjectCreationExpressionSyntaxKind()
-            => SyntaxKind.AnonymousObjectCreationExpression;
+        public CSharpConvertAnonymousTypeToTupleDiagnosticAnalyzer()
+            : base(CSharpSyntaxKindsService.Instance)
```

Comment:
So the assumption here is eventually we will move ISyntaxFactsService and ISyntaxKindsService down to the shared layer where these analyzers moved to CodeStyle layer will still be able to use them? Sounds reasonable to me, but just confirming.

Reply:
Yup.  these are pure-data-singletons.  They can move down very effectively.  There may be some places where they're not pure-data (i.e. they use other imported-services).  but those shoudl be very rare, and can be broken out.

Reply:
👍